pyomo/contrib/pyros/uncertainty_sets.py

```diff
@@ -0,0 +1,1094 @@
+            if isinstance(p, Param) or isinstance(p, _ParamData):
+                p.value = point[idx]
+            else:
+                p.fix(point[idx])
+
+        original_values = list(p.value for p in the_params)
+
+        # === Generate constraint for set
+        set_constraint = self.set_as_constraint(uncertain_params=the_params)
+
+        # === value() returns True if the constraint is satisfied, False else.
+        is_in_set = all(value(con.expr) for con in list(set_constraint.values()))
+
+        # === Revert uncertain_params to their original value and unfix them
+        for idx, p in enumerate(the_params):
+            p.value = original_values[idx]
+            if type(p) is Var or type(p) is _VarData:
+                p.unfix()
+
+        return is_in_set
+
+
+class BoxSet(UncertaintySet):
+    """
+    Hyper-rectangle (a.k.a. "Box")
+    """
+
+    def __init__(self, bounds):
+        """
+        BoxSet constructor
+
+        Args:
+            bounds: A list of tuples providing lower and upper bounds (lb, ub) for each uncertain parameter, in the same order as the 'uncertain_params' required input that is to be supplied to the PyROS solve statement.
+        """
+        # === non-empty bounds
+        if len(bounds) == 0:
+            raise AttributeError("Vector of bounds must be non-empty")
+        # === Real number valued bounds
+        if not all(isinstance(bound, (int, float)) for tup in bounds for bound in tup):
+            raise AttributeError("Bounds must be real numbers.")
+        # === Ensure no bound is None e.g. all are bounded
+        if any(bound is None for tup in bounds for bound in tup):
+            raise AttributeError("All bounds for uncertain parameters must be real numbers, not None.")
+        # === Ensure each tuple has a lower and upper bound
+        if not all(len(b) == 2 for b in bounds):
+            raise AttributeError("Vector of bounds must include a finite lb and ub for each uncertain parameter")
+        # === Ensure each lb <= ub
+        if not all(bound[0] <= bound[1] for bound in bounds):
+               raise AttributeError("Lower bounds must be less than or equal to upper bounds")
+
+        self.bounds = bounds
+        self.type = "box"
+
+    @property
+    def dim(self):
+        return len(self.bounds)
+
+    @property
+    def geometry(self):
+        return 1
+
+    def set_as_constraint(self, uncertain_params, **kwargs):
+        """
+        Function to generate constraints for the BoxSet uncertainty set.
+
+        Args:
+            uncertain_params: uncertain parameter objects for writing constraint objects
+        """
+
+        conlist = ConstraintList()
+        conlist.construct()
+
+        set_i = list(range(len(uncertain_params)))
+
+        for i in set_i:
+            conlist.add(uncertain_params[i] >= self.bounds[i][0])
+            conlist.add(uncertain_params[i] <= self.bounds[i][1])
+
+        return conlist
+
+    @staticmethod
```

Comment:
Because this function doesn't operate on UncertaintySet objects, it is only used in the creation of the separation problem to add numerically derived bounds on each uncertain parameter to help with solver convergence. It seemed like being a static method was more appropriate, in that it is still specific to a particular class of UncertaintySet but doesn't need any information from `self`.